src/main/java/core/basesyntax/Car.java

```diff
@@ -1,5 +1,22 @@
 package core.basesyntax;
 
-public class Car {
+public class Car implements Cloneable {
+    private String producer;
+    private int countOfDoors;
+    private Engine engine;
+    private String color;
+    private String type;
 
+    public Car(String producer, int countOfDoors, Engine engine, String color, String type) {
+        this.producer = producer;
+        this.countOfDoors = countOfDoors;
+        this.engine = engine;
+        this.color = color;
+        this.type = type;
+    }
+
+    @Override
+    protected Object clone() {
```

Comment:
let's change return type from Object to Car, so user, who will use this method won't need to cast it